### PR TITLE
Improve aggregates integration test

### PR DESCRIPTION
**Pull request by imakhlin**

### CODE REVIEW

File: test/integration/aggregates/aggregates_integration_test.go [hunk 1]
```diff
@@ -170,33 +178,30 @@ func testAggregatesCase(t *testing.T, testConfig *TestConfig) {
...
 		if iter.Err() != nil {
 			t.Fatal(iter.Err())
 		}
 
 		if aggr == "count" {
-			fmt.Println("\nTotal count read:", totalCount)
-			if totalCount != total {
-				t.Fatalf("Expected total count of %d, got %d", total, totalCount)
-			}
+			assert.Equal(t, testConfig.expectedCount, totalCount, "count is not as expected")
```
Comment (talIguaz) [inline test/integration/aggregates/aggregates_integration_test.go]: You already checked it on line 183

File: test/integration/aggregates/aggregates_integration_test.go [hunk 2]
```diff
@@ -170,33 +178,30 @@ func testAggregatesCase(t *testing.T, testConfig *TestConfig) {
...
 		}
 
 		fmt.Println()
```
Comment (talIguaz) [inline test/integration/aggregates/aggregates_integration_test.go]: delete

File: test/benchmark/testdata/tsdb-bench-test-config.yaml [hunk 3]
```diff
@@ -34,4 +34,7 @@ AppendOneByOne: true
 BatchSize: 64
 
 # Cleanup the data after each test
-CleanupAfterTest: true
\ No newline at end of file
+CleanupAfterTest: true
+
+# aggregate query step size - used by validators
+QueryAggregateStep: "1h"
```
[the remark this answers is not in the document]
Comment (imakhlin) [reply test/benchmark/testdata/tsdb-bench-test-config.yaml]: yep, by default I want aggregate from the raw data

File: test/integration/aggregates/aggregates_integration_test.go [hunk 4]
```diff
@@ -170,33 +178,30 @@ func testAggregatesCase(t *testing.T, testConfig *TestConfig) {
...
-				if aggr == "sum" && v != expectedSum {
-					t.Errorf("Sum is not ok - expected %f, got %f at time %d", expectedSum, v, tm)
-				}
-				if aggr == "avg" && v != expectedSum/float64(expectedCount) {
-					t.Errorf("Avg is not ok - expected %f, got %f at time %d", expectedSum/float64(expectedCount), v, tm)
-				}
-			}
+			if tm > testConfig.testStartTime && tm < testConfig.testEndTime-testConfig.queryStep {
 
+				assert.Condition(t, func() bool { return aggr == "count" && int64(v) != expectedCount },
```
Comment (talIguaz) [inline test/integration/aggregates/aggregates_integration_test.go]: The error message will be displayed whenever the condition it `false` looks like it is the opposite of what you want, isn't it?

File: pkg/tsdb/v3iotsdb_integration_test.go [hunk 5]
```diff
@@ -187,26 +202,31 @@ func TestQueryData(t *testing.T) {
...
 			labels: utils.FromStrings("os", "linux", "iguaz", "yesplease"),
 			data: []tsdbtest.DataPoint{{Time: 1532940510, Value: 300.3},
 				{Time: 1532940510 + 60, Value: 300.3},
 				{Time: 1532940510 + 2*60, Value: 100.4},
 				{Time: 1532940510 + 2*60, Value: 200.0}},
 			from:        1532940510,
 			to:          1532940510 + 6*60,
+			step:        defaultStepMs,
 			aggregators: "sum",
 			expected:    map[string][]tsdbtest.DataPoint{"sum": {{Time: 1532940510, Value: 901.0}}}},
```
[the remark this answers is not in the document]
Comment (imakhlin) [reply pkg/tsdb/v3iotsdb_integration_test.go]: not in this PR

File: test/integration/aggregates/aggregates_integration_test.go [hunk 6]
```diff
@@ -168,48 +165,49 @@ func testAggregatesCase(t *testing.T, testConfig *TestConfig) {
...
-				assert.Condition(t, func() bool { return aggr == "count" && int64(v) != expectedCount },
-					"Count is not ok - expected %d, got %f at time %d", expectedCount, v, tm)
-
-				assert.Condition(t, func() bool { return aggr == "sum" && v != expectedSum },
-					"Sum is not ok - expected %f, got %f at time %d", expectedSum, v, tm)
-
-				assert.Condition(t, func() bool { return aggr == "avg" && v != expectedSum/float64(expectedCount) },
-					"Avg is not ok - expected %f, got %f at time %d", expectedSum/float64(expectedCount), v, tm)
+			_, v := iter.At()
+			//fmt.Printf("t=%d,v=%f\n", tm, v)
```
Comment (talIguaz) [inline test/integration/aggregates/aggregates_integration_test.go]: delete

File: test/integration/aggregates/aggregates_integration_test.go [hunk 7]
```diff
@@ -168,48 +165,49 @@ func testAggregatesCase(t *testing.T, testConfig *TestConfig) {
 		series := set.At()
 		lset := series.Labels()
 		aggr := lset.Get("Aggregator")
-		fmt.Println("\n\nLables:", lset)
+		//fmt.Printf("\nLables: %v", lset)
```
Comment (talIguaz) [inline test/integration/aggregates/aggregates_integration_test.go]: delete

File: test/integration/aggregates/aggregates_integration_test.go [hunk 8]
```diff
@@ -168,48 +165,49 @@ func testAggregatesCase(t *testing.T, testConfig *TestConfig) {
...
+
+			switch aggr {
+			case "count":
+				actualCount += int(v)
+			case "sum":
+				actualSum += v
+			case "avg":
+				if ! math.IsNaN(v) {
+					actualAvgResultsCount += 1
+					actualAvgResultsSum = actualAvgResultsSum + v
```
Comment (talIguaz) [inline test/integration/aggregates/aggregates_integration_test.go]: Not sure i understand what you count here. is it because there is an AVG per step?

**Pull request merged**